src/suffer.cpp

```diff
@@ -0,0 +1,1967 @@
+        if( !( weapon.has_flag( "RAIN_PROTECT" ) ) ) {
+            add_msg_if_player( m_bad, _( "The sunlight burns your skin!" ) );
+            if( has_effect( effect_sleep ) && !has_effect( effect_narcosis ) ) {
+                wake_up();
+            }
+            mod_pain( 1 );
+            hurtall( 1, nullptr );
+        }
+    }
+
+    if( ( has_trait( trait_TROGLO ) || has_trait( trait_TROGLO2 ) ) &&
+        g->weather.weather == WEATHER_SUNNY ) {
+        mod_str_bonus( -1 );
+        mod_dex_bonus( -1 );
+        add_miss_reason( _( "The sunlight distracts you." ), 1 );
+        mod_int_bonus( -1 );
+        mod_per_bonus( -1 );
+    }
+    if( has_trait( trait_TROGLO2 ) ) {
+        mod_str_bonus( -1 );
+        mod_dex_bonus( -1 );
+        add_miss_reason( _( "The sunlight distracts you." ), 1 );
+        mod_int_bonus( -1 );
+        mod_per_bonus( -1 );
+    }
+    if( has_trait( trait_TROGLO3 ) ) {
+        mod_str_bonus( -4 );
+        mod_dex_bonus( -4 );
+        add_miss_reason( _( "You can't stand the sunlight!" ), 4 );
+        mod_int_bonus( -4 );
+        mod_per_bonus( -4 );
+    }
+}
+
+void Character::suffer_from_albinism()
+{
+    if( !one_turn_in( 1_minutes ) ) {
+        return;
+    }
+    // Sunglasses can keep the sun off the eyes.
+    if( !has_bionic( bionic_id( "bio_sunglasses" ) ) &&
+        !( wearing_something_on( bp_eyes ) &&
+           ( worn_with_flag( "SUN_GLASSES" ) || worn_with_flag( "BLIND" ) ) ) ) {
+        add_msg_if_player( m_bad, _( "The sunlight is really irritating your eyes." ) );
+        if( one_turn_in( 1_minutes ) ) {
+            mod_pain( 1 );
+        } else {
+            focus_pool --;
+        }
+    }
+    // Umbrellas can keep the sun off the skin
+    if( weapon.has_flag( "RAIN_PROTECT" ) ) {
+        return;
+    }
+    //calculate total coverage of skin
+    body_part_set affected_bp { {
+            bp_leg_l, bp_leg_r, bp_torso, bp_head, bp_mouth, bp_arm_l,
+            bp_arm_r, bp_foot_l, bp_foot_r, bp_hand_l, bp_hand_r
+        }
+    };
+    //pecentage of "open skin" by body part
+    std::map<body_part, float> open_percent;
+    //initialize coverage
+    for( const body_part &bp : all_body_parts ) {
+        if( affected_bp.test( bp ) ) {
+            open_percent[bp] = 1.0;
+        }
+    }
+    //calculate coverage for every body part
+    for( const item &i : worn ) {
+        body_part_set covered = i.get_covered_body_parts();
+        for( const body_part &bp : all_body_parts )  {
+            if( !affected_bp.test( bp ) || !covered.test( bp ) ) {
+                continue;
+            }
+            //percent of "not covered skin"
+            float p = 1.0 - i.get_coverage() / 100.0;
+            open_percent[bp] = open_percent[bp] * p;
+        }
+    }
+
+    const float COVERAGE_LIMIT = 0.01;
+    body_part max_affected_bp = num_bp;
+    float max_affected_bp_percent = 0;
+    int count_affected_bp = 0;
+    for( auto &it : open_percent ) {
```

Comment:
could be a good opportunity to remove `auto`

Reply:
Fixed.